vhdbuilder/packer/trivy-scan.sh

```diff
@@ -1,41 +1,142 @@
 tar -xvzf "trivy_${TRIVY_VERSION}_${TRIVY_ARCH}.tar.gz"
 rm "trivy_${TRIVY_VERSION}_${TRIVY_ARCH}.tar.gz"
 chmod a+x trivy 
 
-./trivy --scanners vuln rootfs -f json --skip-dirs /var/lib/containerd --ignore-unfixed --severity HIGH,CRITICAL -o "${TRIVY_REPORT_JSON_PATH}" /
+# pull vuln-to-kusto binary
+az storage blob download --auth-mode login --account-name vulntokustobinary -c vuln-to-kusto \
+    --blob-url https://vulntokustobinary.blob.core.windows.net/vuln-to-kusto/v0.0.1-5a0563d8080/vuln-to-kusto-vhd_linux_${GO_ARCH} \
+    --file ./vuln-to-kusto-vhd
+chmod a+x vuln-to-kusto-vhd
+
+# shellcheck disable=SC2155
+export PATH="$(pwd):$PATH"
+
+./trivy --scanners vuln rootfs -f json --skip-dirs /var/lib/containerd --ignore-unfixed --severity HIGH,CRITICAL -o "${TRIVY_REPORT_ROOTFS_JSON_PATH}" /
+./vuln-to-kusto-vhd scan-report \
+    --vhd-vhdname="${VHD_NAME}" \
+    --vhd-ossku="${OS_SKU}" \
+    --vhd-osversion="${OS_VERSION}" \
+    --vhd-skuname="${SKU_NAME}" \
+    --kusto-endpoint=${KUSTO_ENDPOINT} \
+    --kusto-database=${KUSTO_DATABASE} \
+    --kusto-table=${KUSTO_TABLE} \
+    ${TRIVY_REPORT_ROOTFS_JSON_PATH}
 
 IMAGE_LIST=$(ctr -n k8s.io image list -q | grep -v sha256)
 
 echo "This contains the list of images with high and critical level CVEs (if present), that are present in the node. 
-Note: images without CVEs are also listed" >> "${TRIVY_REPORT_TABLE_PATH}"
+Note: images without CVEs are also listed" >> "${TRIVY_REPORT_IMAGE_TABLE_PATH}"
+
+for CONTAINER_IMAGE in $IMAGE_LIST; do
+    # append to table
+    ./trivy --scanners vuln image --ignore-unfixed --severity HIGH,CRITICAL -f table ${CONTAINER_IMAGE} >> ${TRIVY_REPORT_IMAGE_TABLE_PATH} || true
 
-for image in $IMAGE_LIST; do
-    ./trivy --scanners vuln image --ignore-unfixed --severity HIGH,CRITICAL -f table $image >> ${TRIVY_REPORT_TABLE_PATH} || true
+    # export to Kusto, one by one
+    BASE_CONTAINER_IMAGE=$(basename ${CONTAINER_IMAGE})
+    TRIVY_REPORT_IMAGE_JSON_PATH=${TRIVY_REPORT_DIRNAME}/trivy-report-image-${BASE_CONTAINER_IMAGE}.json
+    ./trivy --scanners vuln image -f json --ignore-unfixed --severity HIGH,CRITICAL -o ${TRIVY_REPORT_IMAGE_JSON_PATH} $CONTAINER_IMAGE || true
+
+    if [[ -f ${TRIVY_REPORT_IMAGE_JSON_PATH} ]]; then
+        ./vuln-to-kusto-vhd scan-report \
+            --vhd-vhdname="${VHD_NAME}" \
+            --vhd-ossku="${OS_SKU}" \
+            --vhd-osversion="${OS_VERSION}" \
+            --vhd-skuname="${SKU_NAME}" \
+            --kusto-endpoint=${KUSTO_ENDPOINT} \
+            --kusto-database=${KUSTO_DATABASE} \
+            --kusto-table=${KUSTO_TABLE} \
+            ${TRIVY_REPORT_IMAGE_JSON_PATH} || true
+
+        rm ${TRIVY_REPORT_IMAGE_JSON_PATH} || true
+    fi
 done
 
 rm ./trivy 
 
-chmod a+r "${TRIVY_REPORT_JSON_PATH}"
-chmod a+r "${TRIVY_REPORT_TABLE_PATH}"
+chmod a+r "${TRIVY_REPORT_ROOTFS_JSON_PATH}"
```

Comment:
ain't you removing the report on line `123` ?

Reply:
So we 
- remove  `trivy json` output for images after uploading
- keep `trivy rootfs` output + `trivy table` output, for uploading to storage account and leaving them as pipeline artifacts.